index.js

```diff
@@ -100,185 +113,209 @@ const VerifyOptions = t.interface({
+/**
+ * The refresh token has expired or was not found
+ * @type {StandardError}
+ * @typedef RefreshTokenExpiredOrNotFound
+ * @property {String} [name='RefreshTokenExpiredOrNotFound']
+ */
+const refreshTokenExpiredOrNotFound = new StandardError(
+  'The refresh token has expired or was not found',
+  {name: 'RefreshTokenExpiredOrNotFound'}
+);
 
 /**
- * Token pairs
- * @typedef Tokens
- * @type {Object}
- * @property {String} accessToken
- * @property {Number} accessTokenExpiresAt epoch
- * @property {String} refreshToken
- * @property {Number} refreshTokenExpiresAt epoch
+ * The access token provided is invalid
+ * @type {StandardError}
+ * @typedef InvalidAccessToken
+ * @property {String} [name='InvalidAccessToken']
  */
+const invalidAccessToken = new StandardError(
+  'The access token provided is invalid',
+  {name: 'InvalidAccessToken'}
+);
 
-const _getTokensObj = (accessToken, accessTokenExpiresAt, refreshToken, refreshTokenExpiresAt) => ({
-  accessToken,
-  accessTokenExpiresAt,
-  refreshToken,
-  refreshTokenExpiresAt
-});
+// 15 minutes
+const regularAccessTTL = 60 * 15;
+// 1 hour
+const prolongedAccessTTL = 60 * 60;
+// 25 minutes
+const regularRefreshTTL = 60 * 25;
+// 4 days
+const prolongedRefreshTTL = 60 * 60 * 24 * 4;
 
-class Authomatic {
-  /**
-   * Constructor
-   * @param {Object} store
-   * @param {string} [algorithm=HS256] algorithm cannot be 'none'
-   * @param {Number} [expiresIn=60 * 30] expiration time in seconds.
-   * @param {Object} [jwt] jsonwebtoken instance, by default it uses require('jsonwebtoken')
-   * @param {UserSignOptions} [defaultSignInOptions]
-   * @param {VerifyOptions} [defaultVerifyOptions]
-   */
-  constructor({
-    store, algorithm = 'HS256', expiresIn = regularTokenLifeInSeconds, jwt = jsonwebtoken,
-    defaultSignInOptions = {}, defaultVerifyOptions = {}
-  }) {
-    this._store = Store(store);
-    this._defaultSignInOptions = UserSignOptions(defaultSignInOptions);
-    this._defaultVerifyOptions = VerifyOptions(defaultVerifyOptions);
-    this._algorithm = Algorithm(algorithm);
-    this._expiresIn = ExpiresIn(expiresIn);
-    this._jwt = JWT(jwt);
-  }
+// Seconds -> Seconds Since the Epoch
+const computeExpiryDate = seconds => Math.floor(Date.now() / 1000) + seconds;
 
-  /**
-   * @private
-   * A private function that creates a refresh token
-   * @param {String|Number} userId
-   * @param {String} accessToken
-   * @param {Number} ttl time to live in milliseconds
-   * @returns {Promise}
-   */
-  async _createRefreshToken(userId, accessToken, ttl) {
-    const refreshToken = generator.generate(256);
-    await this._store.registerTokens(userId, refreshToken, accessToken, ttl);
-    return refreshToken;
-  }
+const generateRefreshToken = userId => {
+  const refreshToken = {rt: generator.generate(256), uid: userId};
+  const stringifiedRefreshToken = JSON.stringify(refreshToken);
+  return Base64.encode(stringifiedRefreshToken);
```

Comment:
Since you are base64 encoding it, you don't really need the `generator` module, I think. You can just do something like:

```js
userId => `${userId}_${crypto.randomBytes(256).toString('base64')}`
```

This should also simplify the store logic, as it can now use the refresh token as a key directly.

Furthermore, I think 256 bytes of random entropy is overkill, and increases our odds of depleting the random source under heavy load.

The downside is that we are treating `UserId` as a String now, which is something to consider.